includes/filerepo/backend/GcsFileBackend.php

```diff
@@ -112,21 +300,35 @@ protected function doDeleteInternal( array $params ) {
 	protected function doGetFileStat( array $params ) {
 		WikiaLogger::instance()->info( __METHOD__, [
 			'call_stack' => ( new Exception() )->getTraceAsString(),
-			'storage_path' => $params['src']
+			'storage_path' => $params['src'],
 		] );
 
 		$obj = $this->bucket()->object( $this->objectName( $params['src'] ) );
 
 		if ( $obj->exists() ) {
 			return [
 				'mtime' => wfTimestamp( TS_MW, $obj->info()['updated'] ),
-				'size' => $obj->info()['size']
+				'size' => $obj->info()['size'],
+				'sha1' => $this->sha1ToHash( $obj->info()['sha1'] ),
 			];
 		} else {
 			return false;
 		}
 	}
 
+	private function getMetadata( $sha1 ) {
+		return [
+			// user metadata are nested like so: metadata.metadata
+			'metadata' => [
+				'sha1' => $sha1,
+			]
+		];
+	}
+
+	private function sha1ToHash( $sha1 ) {
```

Comment:
I don't see where this is used

Reply:
https://github.com/Wikia/app/pull/16732/files#diff-dee15708210663e64e57b83e68d93219R312